boxoffice/forms/discount.py

```diff
@@ -0,0 +1,99 @@
+
+__all__ = ['DiscountPolicyForm', 'PriceBasedDiscountPolicyForm', 'DiscountPriceForm', 'DiscountCouponForm', 'AutomaticDiscountPolicyForm', 'CouponBasedDiscountPolicyForm']
+
+
+class DiscountPolicyForm(forms.Form):
+    title = forms.StringField(__("Discount title"),
+        validators=[forms.validators.DataRequired(__("Please specify a discount title")),
+        forms.validators.Length(max=250)], filters=[forms.filters.strip()])
+    discount_type = forms.RadioField(__("Discount type"),
+        choices=DISCOUNT_TYPE.items(), coerce=int, default=DISCOUNT_TYPE.COUPON)
+    is_price_based = forms.RadioField(__("Price based discount"),
+        coerce=getbool, default=True, choices=[
+            (True, __("Special price discount")),
+            (False, __("Percentage based discount"))])
+
+
+def validate_unique_discount_code_base(form, field):
+    if DiscountPolicy.query.filter(DiscountPolicy.id != form.edit_id, DiscountPolicy.discount_code_base == field.data).notempty():
+        raise StopValidation(__("This discount coupon prefix already exists. Please pick a different prefix"))
+
+
+class AutomaticDiscountPolicyForm(DiscountPolicyForm):
+    item_quantity_min = forms.IntegerField(__("Minimum number of tickets"), default=1)
+    percentage = forms.IntegerField(__("Percentage"),
+        validators=[forms.validators.DataRequired(__("Please specify a discount percentage"))])
+    items = QuerySelectMultipleField(__("Items"), get_label='title',
+        validators=[forms.validators.DataRequired(__("Please select at least one item for which the discount is applicable"))])
+
+    def set_queries(self):
+        self.items.query = Item.query.join(ItemCollection).filter(
+            ItemCollection.organization == self.edit_parent).options(db.load_only('id', 'title'))
+
+
+class CouponBasedDiscountPolicyForm(DiscountPolicyForm):
+    items = QuerySelectMultipleField(__("Items"), get_label='title',
+        validators=[forms.validators.DataRequired(__("Please select at least one item for which the discount is applicable"))])
+    percentage = forms.IntegerField(__("Percentage"),
+        validators=[forms.validators.DataRequired(__("Please specify a discount percentage")),
+            forms.validators.NumberRange(min=1, max=100, message=__("Percentage should be between >= 1 and <= 100"))])
+    discount_code_base = forms.StringField(__("Discount title"),
+        validators=[forms.validators.DataRequired(__("Please specify a discount code base")),
+        forms.validators.Length(max=20), validate_unique_discount_code_base],
+        filters=[forms.filters.strip(), forms.filters.none_if_empty()])
+    bulk_coupon_usage_limit = forms.IntegerField(__("Bulk coupon usage limit"), default=1)
+
+    def set_queries(self):
+        self.items.query = Item.query.join(ItemCollection).filter(
+            ItemCollection.organization == self.edit_parent).options(db.load_only('id', 'title'))
```

Comment:
It's actually in a similar format i.e `item_collection_title: item_title`. `select2` has been configured to hit a different endpoint for the titles and ids - https://github.com/hasgeek/boxoffice/blob/admin-discount/boxoffice/views/admin_item.py#L28.